### PR TITLE
Allow leading zeros for NumberedFileInputSplit 

### PR DESCRIPTION
**Pull request by turambar**
## What changes were proposed in this pull request?

With these changes, NumberedFileInputSplit permits two kinds of formatting -- "prefix4.suffix" or "prefix0004.suffix." Latter is more suitable for, e.g., Hadoop mapfiles spit out by a Spark ETL pipeline.

## How was this patch tested?

Added a NumberedFileInputSplitTests class with some unit tests to verify the name formatting works. I've also run it in live code, and it performed fine.

### CODE REVIEW

**Comment by turambar** [comment]
Addresses this issue: https://github.com/deeplearning4j/DataVec/issues/421

File: datavec-api/src/main/java/org/datavec/api/split/NumberedFileInputSplit.java [hunk 1]
```diff
@@ -1,19 +1,3 @@
-/*-
```
Comment (huitseeker) [inline datavec-api/src/main/java/org/datavec/api/split/NumberedFileInputSplit.java]: This should not be removed

File: datavec-api/src/main/java/org/datavec/api/split/NumberedFileInputSplit.java [hunk 2]
```diff
@@ -39,15 +25,18 @@
     private final int minIdx;
     private final int maxIdx;
 
+    private static final Pattern p = Pattern.compile("\\%(0\\d)?d");
```
Comment (huitseeker) [inline datavec-api/src/main/java/org/datavec/api/split/NumberedFileInputSplit.java]: To be compliant with `printf`-formatted strings, which is the format your users will expect to deal with, you have to at least document the following interactions in unit tests (and possibly do something better in the cases below)

```
scala> val pat = "\\%(0\\d)?d".r
pat: scala.util.matching.Regex = \%(0\d)?d

scala> def f(x: String): String = x match {
     |   case pat.unanchored(y) => s"Matched, with pattern $y"
     |   case _ => "UNMATCHED"
     | }
f: (x: String)String

scala> f("/path/to/files/prefix%5d.suffix")
res24: String = UNMATCHED

scala> f("/path/to/files/prefix%+5d.suffix")
res26: String = UNMATCHED

scala> f("/path/to/files/prefix%-5d.suffix")
res27: String = UNMATCHED

scala> f("/path/to/files/prefix%011d.suffix")
res28: String = UNMATCHED

scala> f("/path/to/files/prefix%101d.suffix")
res29: String = UNMATCHED

scala> f("/path/to/files/prefix%0505d.suffix")
res30: String = UNMATCHED
```

File: datavec-api/src/test/java/org/datavec/api/split/NumberedFileInputSplitTests.java [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    @Test
+    public void testNumberedFileInputSplitWithLeadingZeroesNoSuffix() {
+        String baseString = "/path/to/files/prefix-%d";
+        int minIdx = 0;
+        int maxIdx = 10;
+        runNumberedFileInputSplitTest(baseString, minIdx, maxIdx);
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testNumberedFileInputSplitWithLeadingSpaces() {
```
Comment (huitseeker) [inline datavec-api/src/test/java/org/datavec/api/split/NumberedFileInputSplitTests.java]: O_o

**Pull request merged**